synapse/lib/cell.py

```diff
@@ -4342,3 +4348,62 @@ async def _onUserDelEvnt(self, evnt):
                 self.slab.delete(key_iden, db=self.apikeydb)
             self.slab.delete(lkey, db=self.usermetadb)
             await asyncio.sleep(0)
+
+    def _makeCachedSslCtx(self, opts):
+
+        opts = dict(opts)
+
+        cadir = self.conf.get('tls:ca:dir')
+
+        purpose = ssl.Purpose[opts['purpose']]
```

Comment:
Having just working on replacing one library for another; can we add a layer with this purpose key so we're not directly tied to the core Python `ssl.Purpose` values? maybe a dictionary that maps "our" values to the SSL purpose?

Reply:
I decided to remove this from the schema altogether since I didn't plan on accepting other values yet.  I say we cross that bridge when we have the need.